src/components/Discussion/Composer/CommentComposer.js

```diff
@@ -102,22 +130,37 @@ export const CommentComposer = props => {
    */
   const localStorageKey = commentComposerStorageKey(id)
 
-  const [text, setText] = React.useState(() => {
-    if (props.initialText) {
-      return props.initialText
-    } else if (typeof localStorage !== 'undefined') {
-      try {
-        return localStorage.getItem(localStorageKey) || ''
-      } catch (e) {
-        return ''
-      }
-    } else {
-      return ''
+  const fetchPreview = text => {
+    if (!actions || !actions.previewComment) {
+      return setPreview(null)
     }
-  })
+    textRef.current = text
+    actions
+      .previewComment({
+        content: text,
+        discussionId: id,
+        parentId,
+        id: commentId
+      })
+      .then(nextPreview => {
+        if (textRef.current === text) {
+          setPreview(nextPreview)
+        }
+      })
+      .catch(() => {
+        if (textRef.current === text) {
+          setPreview(null)
+        }
+      })
+  }
+
+  React.useEffect(() => {
+    fetchPreview(text)
+  }, [text])
```

Comment:
We should probably debounce this by 100ms or so. See `republik-frontend/components/FontSize/Sync.js` for an example.